Sources/OpenAPIRuntime/EventStreams/ServerSentEventsDecoding.swift

```diff
@@ -110,6 +139,19 @@ extension AsyncSequence where Element == ArraySlice<UInt8>, Self: Sendable {
                 )
             }
     }
+
+    public func asDecodedServerSentEventsWithJSONData<JSONDataType: Decodable>(
+        of dataType: JSONDataType.Type = JSONDataType.self,
+        decoder: JSONDecoder = .init(),
+        terminatingData: ArraySlice<UInt8>
```

Comment:
`lastElement` or `stopOn` are other alternatives

Reply:
I think we need to name this to indicate that it's cutting the sequence off. In practice it will be used to detect what will be the last element anyway. 

I'm actually wondering if we should try and fit more into the existing nomenclature here. e.g. could this be called `while:` (like `prefix(while:)`). I think that's pretty defensible and clear. The semantics are _while this is true, decode the events as JSON data_.

Reply:
I see, and in this case, the bool would be flipped, so like this?

```swift
response.asDecodedServerSentEventsWithJSONData(of: ..., while: { $0 != ArraySlice("[DONE]".utf8) })
```

Yeah I like that, it allows us to just reinforce conventions from the standard library rather than invent our own.

Reply:
`while:` is perfect!

But we haven't settled on a name for what is currently called `terminatingData:`. Maybe `untilReceived:`? Can you think of something that fits in well with existing standard library nomenclature here as well?

Reply:
Is this for the overload that just takes the String/Data? I'm not sure about how to make that clear. Do we need to provide it all?

Reply:
Yes, it's for the overload. @czechboy0, I believe you had suggested providing an additional overload initially. What do you reckon?

Reply:
Yeah I suggested the overload, but I also don't have a good idea for the name, so let's hold off on adding it for now, and let's just do the closure-based variant for now?

Reply:
I'd lean towards a single function that has a non-optional closure with a default value of `{ true }`. Basing this on e.g. `print(..., terminator:)`, which is also a non-optional defaulted value.

Reply:
Pushed. Does that work for you, @simonjbeaumont?

Reply:
Yep, I think I'd like the non-optional parameter here. When it was an early-terminator, I think optional made sense. Now it's essentially a filter, I think non-optional default to `{ true }` is better, unless there's no performance downsides.